lib/src/builder/codegen/accessors_generator.dart

```diff
@@ -198,6 +203,12 @@ abstract class TypedMapAccessorsGenerator extends BoilerplateDeclarationGenerato
           annotationCount++;
           isRequired = true;
           isPotentiallyNullable = true;
+
+          if (type.isProps && disableRequiredPropValidation == null) {
+            requiredPropChecks.add('  if(!props.containsKey($keyValue)) {\n'
+                '  throw MissingRequiredPropsError(\'Required prop `$accessorName` is missing.\');\n'
```

Comment:
This needs to be escaped in case the name of the prop has a `$` character in it. (In general it's a good idea to always escape strings by default.)

You can use `stringLiteral` for that here:
```suggestion
                '  throw MissingRequiredPropsError(${stringLiteral('Required prop `$accessorName` is missing.')});\n'
```